File: golden/ruby/1_count_of_largest_integer.rb
def solution(a)
  return 0 if a.size.zero?

  max_int = a[0]
  count = 0

  a.each do |current|
    next count += 1 if max_int == current

    if current > max_int
      max_int = current
      count = 1
    end
  end

  count
end

File: golden/javascript/1_count_of_largest_integer.js
function solution(A) {
    if (!A.length) {
        return 0;
    }

    var maxInt = A[0];
    var count = 1;

    for (var i = 1; i < A.length; i++) {
        if (A[i] > maxInt) {
            maxInt = A[i];
            count = 1;
        }
        else if (A[i] == maxInt) {
            count += 1;
        }
    }

    return count;
}
